atomix/cluster/src/test/java/io/atomix/raft/partition/impl/RaftPartitionServerTest.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.atomix.raft.partition.impl;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import io.atomix.cluster.ClusterMembershipService;
+import io.atomix.cluster.MemberId;
+import io.atomix.cluster.messaging.ClusterCommunicationService;
+import io.atomix.primitive.partition.PartitionId;
+import io.atomix.primitive.partition.PartitionMetadata;
+import io.atomix.raft.partition.RaftPartition;
+import io.atomix.raft.partition.RaftPartitionConfig;
+import io.atomix.raft.partition.RaftPartitionGroupConfig;
+import java.time.Duration;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import org.junit.jupiter.api.Test;
+
+public class RaftPartitionServerTest {
+
+  private final RaftPartition raftPartition = mock(RaftPartition.class);
+  private final RaftPartitionGroupConfig partitionGroupConfig =
+      mock(RaftPartitionGroupConfig.class);
+
+  @Test
+  public void testInitServerRuntimeExceptionReturnsExceptionalFuture() {
```

Comment:
this is not really pretty as it depends on much internals of the RaftPartitionServer setup but I couldn't find an easier straightforward way to test this behavior